### ISSUE TITLE
Remove placeholder for date input

### ISSUE THREAD

**Issue by kata-lin** [Legal-and-General/canopy#1363]
The placeholder for the date input doesn't have any styling added especially regarding colour. Due to this the user agent stylesheet is being applied and in Safari's case the contrast ratio fails from accessibility point of view.

Since the component already has a hint, the best option is to be able to toggle or completely remove the placeholder.

**Comment by pauleustice** [comment]
@pauleustice To raise a ticket for this work

**Comment by pauleustice** [comment]
Hi @kata-lin! I've investigated this and think the best option is to simply apply a specific colour to the placeholder shadow DOM element for the input.

Using `color-charcoal` (#333) makes it look like this:

<img width="1383" alt="Screenshot 2024-05-07 at 12 34 48" src="https://github.com/Legal-and-General/canopy/assets/1677737/8fc02820-5bdd-441e-97ae-e31e30c7f3a8">

That's the same colour as the main input text, so will be accessible in terms of contrast.

**Comment by kata-lin** [comment]
Yes, that sounds good @pauleustice. Thank you.
